Sources/StreamChatSwiftUI/ChatChannel/Composer/MessageComposerViewModel+Recording.swift

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import SwiftUI
+
+public struct AudioRecordingInfo: Equatable {
+    /// The waveform of the recording.
+    public var waveform: [Float]
+    /// The duration of the recording.
+    public var duration: TimeInterval
+    
+    mutating func update(with entry: Float, duration: TimeInterval) {
```

Comment:
Why is this one needed? The properties are `var` so probably we can update them in calling place.

Reply:
It's possible to do it in place, but I think it's cleaner with a method.